[Merged by Bors] - Clarify description of API docs (#425)
bevyengine/bevy-website: content/learn/book/next-steps/_index.md

* **[The Bevy Examples](https://github.com/bevyengine/bevy/tree/latest/examples#examples)**: We create an example for every major Bevy feature. This is currently the best way to learn Bevy's features and how to use them.
  * **[The Bevy Web Examples](https://bevyengine.org/examples)**: We also try to make most of these examples available directly in your browser.
* **[Breakout](https://github.com/bevyengine/bevy/blob/latest/examples/games/breakout.rs)**: A small "breakout" clone that illustrates what a real Bevy game looks like.

> * **[Bevy API Docs](https://docs.rs/bevy)**: Documentation and examples for each type, trait and method in Bevy.

Reviewer: ```suggestion
* **[Bevy API Docs](https://docs.rs/bevy)**: Documentation with examples for each type, trait and method in Bevy.
```

Examples are part of the documentation.

Author: Somehow the addition of "with" now messes with how the sentence is perceived, like if "for" related to only examplese:
```
(Documentation and examples) for (each type, trait and method) ...
```
vs
``` 
Documentation with (examples for types, traits, methods and other items) ...
```

Reviewer: IMO the ambiguity still remains with “and”.  Also, the reader is able to trivially resolve it.

Reviewer 2: I like "with" better too :)

> * **[Bevy API Docs](https://docs.rs/bevy)**: Bevy's Rust API reference documentation which includes both auto-generated entries and hand-written explanations.

Reviewer 2: ```suggestion
* **[Bevy API Docs](https://docs.rs/bevy)**: Documentation for each type and trait used in Bevy.
```

I don't think that including either "Rust" or "API" is useful here, but it's important to communicate that it's granular.
